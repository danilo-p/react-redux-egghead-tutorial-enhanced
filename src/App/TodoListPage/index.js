import React from 'react';
import { Header, Container, Grid } from 'semantic-ui-react';
import TodoList from './TodoList';
import styles from './index.styl';

/**
 * Describes the todo list page.
 *
 * @returns {jsx} The TodoListPage template
 */
export default function TodoListPage() {
  return (
    <Container>
      <Grid centered>
        <Grid.Column computer="8" tablet="8" mobile="16">
          <Header as="h1" className={styles.TodoListPage__Header}>
            A Simple Todo App
          </Header>
          <TodoList />
        </Grid.Column>
      </Grid>
    </Container>
  );
}
