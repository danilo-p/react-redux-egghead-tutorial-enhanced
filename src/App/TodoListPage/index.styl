.TodoApp__Header
  margin-top: 1em !important;
  margin-bottom: 1em !important;