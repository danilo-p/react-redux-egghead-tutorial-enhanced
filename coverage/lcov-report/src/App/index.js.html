<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/App/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/App</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/7</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React <span class="cstat-no" title="statement not covered" >from 'react';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span>
import { crea<span class="cstat-no" title="statement not covered" >teStore } from 'r</span>edux';
import { Provider <span class="cstat-no" title="statement not covered" >} from 'react-redux';</span>
import TodoListPage <span class="cstat-no" title="statement not covered" >from './TodoListPage';<span class="cstat-no" title="statement not covered" ></span></span>
import rootRe<span class="cstat-no" title="statement not covered" >ducer from '../common/ducks/index</span>';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
&nbsp;
/**
* SFC that describes the app.
* @see SFC https://reactjs.org/docs/components-and-props.html#functional-and-class-components
*
* @export
* @returns {jsx} The TodoApp template
*/
const App <span class="cstat-no" title="statement not covered" >= () =&gt; (<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  &lt;Provider store={createStore(rootReducer)}&gt;
    &lt;TodoListPage /&gt;<span class="cstat-no" title="statement not covered" ></span>
  &lt;/Provider&gt;
);
&nbsp;
export default App;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jan 01 2018 21:17:22 GMT-0200 (-02)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
