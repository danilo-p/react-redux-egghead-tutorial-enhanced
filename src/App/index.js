import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import TodoListPage from './TodoListPage';
import rootReducer from '../common/ducks/index';

/**
* SFC that describes the app.
* @see SFC https://reactjs.org/docs/components-and-props.html#functional-and-class-components
*
* @export
* @returns {jsx} The TodoApp template
*/
const App = () => (
  <Provider store={createStore(rootReducer)}>
    <TodoListPage />
  </Provider>
);

export default App;
