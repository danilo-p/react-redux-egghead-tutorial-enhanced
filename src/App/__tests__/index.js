import React from 'react';
import basicComponentTests from 'tests/basicComponentTests';
import App from '../';

describe('App', () => {
  basicComponentTests(<App />);
});
