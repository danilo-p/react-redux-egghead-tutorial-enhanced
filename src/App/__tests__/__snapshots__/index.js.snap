// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <TodoListPage />
</Provider>
`;
